ledger/src/shred.rs

```diff
@@ -975,11 +989,13 @@ pub fn should_discard_shred(
                 return true;
             }
         }
-        ShredVariant::MerkleCode(_) => {
-            stats.num_shreds_merkle_code = stats.num_shreds_merkle_code.saturating_add(1);
+        ShredVariant::MerkleCode(_, chained) => {
+            stats.num_shreds_merkle_code[usize::from(chained)] =
+                stats.num_shreds_merkle_code[usize::from(chained)].saturating_add(1);
```

Comment:
might be easier to read if we have separate stats fields for w/ and w/o chained

Reply:
done.